python/cuml/ensemble/randomforest_common.pyx

```diff
@@ -0,0 +1,448 @@
+                 'split_algo', 'split_criterion', 'min_rows_per_node',
+                 'min_impurity_decrease',
+                 'bootstrap', 'bootstrap_features',
+                 'verbose', 'rows_sample',
+                 'max_leaves', 'quantile_per_tree']
+
+    def __init__(self, split_criterion, seed=None,
+                 n_streams=8, n_estimators=100,
+                 max_depth=16, handle=None, max_features='auto',
+                 n_bins=8, split_algo=1, bootstrap=True,
+                 bootstrap_features=False,
+                 verbose=False, min_rows_per_node=2,
+                 rows_sample=1.0, max_leaves=-1,
+                 accuracy_metric=None, dtype=None,
+                 output_type=None, min_samples_leaf=None,
+                 min_weight_fraction_leaf=None, n_jobs=None,
+                 max_leaf_nodes=None, min_impurity_decrease=0.0,
+                 min_impurity_split=None, oob_score=None,
+                 random_state=None, warm_start=None, class_weight=None,
+                 quantile_per_tree=False, criterion=None):
+
+        if accuracy_metric:
+            BaseRandomForestModel.variables.append('accuracy_metric')
+
+        sklearn_params = {"criterion": criterion,
+                          "min_samples_leaf": min_samples_leaf,
+                          "min_weight_fraction_leaf": min_weight_fraction_leaf,
+                          "max_leaf_nodes": max_leaf_nodes,
+                          "min_impurity_split": min_impurity_split,
+                          "oob_score": oob_score, "n_jobs": n_jobs,
+                          "random_state": random_state,
+                          "warm_start": warm_start,
+                          "class_weight": class_weight}
+
+        for key, vals in sklearn_params.items():
+            if vals:
+                raise TypeError(" The Scikit-learn variable ", key,
+                                " is not supported in cuML,"
+                                " please read the cuML documentation for"
+                                " more information")
+
+        if ((seed is not None) and (n_streams != 1)):
+            warnings.warn("For reproducible results in Random Forest"
+                          " Classifier or for almost reproducible results"
+                          " in Random Forest Regressor, n_streams==1 is "
+                          "recommended. If n_streams is > 1, results may vary "
+                          "due to stream/thread timing differences, even when "
+                          "random_seed is set")
+        if handle is None:
+            handle = Handle(n_streams)
+
+        super(BaseRandomForestModel, self).__init__(
+            handle=handle,
+            verbose=verbose,
+            output_type=output_type)
+
+        if max_depth < 0:
+            raise ValueError("Must specify max_depth >0 ")
+
+        self.split_algo = split_algo
+        criterion_dict = {'0': GINI, '1': ENTROPY, '2': MSE,
```

Comment:
This would be better defined outside the init for clarity and easy of modifications in the future